pkg/machine/qemu/virtiofsd.go

```diff
@@ -0,0 +1,116 @@
+//go:build linux || freebsd
+
+package qemu
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v5/pkg/machine/define"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
+	"github.com/containers/storage/pkg/fileutils"
+	"github.com/sirupsen/logrus"
+)
+
+// The preferred mechanism for configuring virtiofsd in an OS component/package
+// is to create a symlink from podman's helper directory to the desired binary.
+// This is how e.g. `catatonit` works.  However for now, we hardcode a few
+// well-known distribution paths here as a fallback.
+const virtiofsdPathFedora = "/usr/libexec/virtiofsd"
+const virtiofsdPathDebian = "/usr/lib/qemu/virtiofsd"
+const virtiofsdPathArch = "/usr/lib/virtiofsd"
+
+// VirtiofsdSpawner spawns an instance of virtiofsd
+type virtiofsdSpawner struct {
+	runtimeDir *define.VMFile
+	binaryPath string
+	mountCount uint
+}
+
+func newVirtiofsdSpawner(runtimeDir *define.VMFile) (*virtiofsdSpawner, error) {
+	var binaryPath string
+	cfg, err := config.Default()
+	if err != nil {
+		return nil, err
+	}
+	binaryPath, err = cfg.FindHelperBinary("virtiofsd", true)
+	if err != nil {
+		logrus.Debugf("Failed to find virtiofsd in helper dir, using fallback")
+		for _, p := range []string{virtiofsdPathDebian, virtiofsdPathFedora, virtiofsdPathArch} {
+			if err := fileutils.Exists(p); err == nil {
+				binaryPath = p
+				break
+			}
+		}
+	}
+	if binaryPath == "" {
+		return nil, fmt.Errorf("failed to find virtiofsd")
```

Comment:
please return the original error from FindHelperBinary because that actually tells the user how to configure it in case virtiofsd is installed.